src/components/ReleaseMonitor/ReleaseMonitorListView.tsx

```diff
@@ -0,0 +1,208 @@
+    component: unparsedFilters.component ? (unparsedFilters.component as string[]) : [],
+  });
+
+  const [activeSortIndex, setActiveSortIndex] = React.useState<number>(
+    SortableHeaders.completionTime,
+  );
+  const [activeSortDirection, setActiveSortDirection] = React.useState<SortByDirection>(
+    SortByDirection.desc,
+  );
+
+  const ReleasesListHeader = React.useMemo(
+    () =>
+      getReleasesListHeader(activeSortIndex, activeSortDirection, (_, index, direction) => {
+        setActiveSortIndex(index);
+        setActiveSortDirection(direction);
+      }),
+    [activeSortDirection, activeSortIndex],
+  );
+
+  const { name, status, application, releasePlan, namespace, component } = filters;
+
+  const { namespaces, namespacesLoaded: loaded } = useNamespaceInfo();
+
+  const [releases, setReleases] = React.useState<MonitoredReleaseKind[]>([]);
+  const [loading, setLoading] = React.useState(true);
+  const [error, setError] = React.useState<unknown>();
+
+  const releasesRef = React.useRef<Record<string, MonitoredReleaseKind[]>>({});
+  const loadedNamespacesRef = React.useRef<Set<string>>(new Set());
+
+  const handleReleasesLoaded = React.useCallback(
+    (ns: string, data: MonitoredReleaseKind[]) => {
+      releasesRef.current[ns] = data;
+      loadedNamespacesRef.current.add(ns);
+
+      if (loadedNamespacesRef.current.size === namespaces.length) {
+        const allReleases = Object.values(releasesRef.current).flat();
+        setReleases(allReleases);
+        setLoading(false);
+        setError(null);
+      }
+    },
+    [namespaces.length],
+  );
+
+  const handleError = React.useCallback((err: unknown) => {
+    setError(err);
+    setLoading(false);
+  }, []);
+
+  React.useEffect(() => {
+    if (loaded && namespaces.length > 0) {
+      setLoading(true);
+      releasesRef.current = {};
+      loadedNamespacesRef.current = new Set();
+    } else if (loaded) {
+      setLoading(false);
+    }
+  }, [loaded, namespaces]);
+
+  const sortedMonitoredReleases = useSortedResources(
+    releases,
+    activeSortIndex,
+    activeSortDirection,
+    sortPaths,
+  );
+
+  const filterOptions = React.useMemo(() => {
+    const apps = new Set<string>();
+    const plans = new Set<string>();
+    const comps = new Set<string>();
+    const nsKeys = namespaces.map((ns) => ns.metadata.name);
+
+    releases.forEach((r) => {
+      const app = r.metadata.labels?.[PipelineRunLabel.APPLICATION];
+      if (app) apps.add(app);
+      const plan = r.spec.releasePlan;
+      if (plan) plans.add(plan);
+      const comp = r.metadata.labels?.[PipelineRunLabel.COMPONENT];
+      if (comp) comps.add(comp);
+    });
+
+    return {
+      statusOptions: createFilterObj(releases, (mr) => getReleaseStatus(mr), statuses),
+      applicationOptions: createFilterObj(
+        releases,
+        (mr) => mr?.metadata.labels[PipelineRunLabel.APPLICATION],
+        [...apps],
```

Comment:
Please remove `[...apps]`, The third argument is not needed here because you are building the options from data.